auth/client/iam/openid4vp.go

```diff
@@ -133,14 +133,32 @@ func (c *OpenID4VPClient) AuthorizationServerMetadata(ctx context.Context, webdi
 	return metadata, nil
 }
 
-func (c *OpenID4VPClient) RequestObject(ctx context.Context, requestURI string) (string, error) {
+func (c *OpenID4VPClient) RequestObject(ctx context.Context, requestURI, requestURIMethod string, walletMetadata *oauth.AuthorizationServerMetadata) (string, error) {
 	iamClient := c.httpClient
 	parsedURL, err := core.ParsePublicURL(requestURI, c.strictMode)
 	if err != nil {
 		return "", fmt.Errorf("invalid request_uri: %w", err)
 	}
-	// the wallet/client acts as authorization server
-	requestObject, err := iamClient.RequestObject(ctx, parsedURL.String())
+
+	var requestObject string
+	switch requestURIMethod {
+	case "", "get":
+		// the wallet/client acts as authorization server
+		requestObject, err = iamClient.RequestObject(ctx, parsedURL.String())
+	case "post":
+		// TODO: consider adding a 'wallet_nonce'
+		form := url.Values{}
+		if walletMetadata != nil {
+			metadataBytes, err := json.Marshal(*walletMetadata)
```

Comment:
POST without metadata would be very weird, as POST is intended to convey information about the wallet (thus metadata?)? (might be my lack of knowledge)

Reply:
it is optional per spec, but we will always send metadata so changed to require metadata.